db/revision_cache_interface.go

```diff
@@ -31,14 +31,14 @@ type RevisionCache interface {
 	// Get returns the given revision, and stores if not already cached.
 	// When includeBody=true, the returned DocumentRevision will include a mutable shallow copy of the marshaled body.
 	// When includeDelta=true, the returned DocumentRevision will include delta - requires additional locking during retrieval.
-	Get(ctx context.Context, docID, revID string, includeBody bool, includeDelta bool) (DocumentRevision, error)
+	Get(ctx context.Context, docID, revID string, cv *CurrentVersionVector, includeBody, includeDelta bool) (DocumentRevision, error)
```

Comment:
I think it would be preferable to define separate APIs for rev- and version-based retrieval from the revision cache, for two reasons:
- it avoids issues around priority (or error handling) if a caller populates both parameters
- it makes it clearer to identify whether the calling code is working with revisions or versions (which will simplify removal in future)